site/content/blog/cue-and-carvel.md

```diff
@@ -0,0 +1,423 @@
+      # try using your own Git repo with a latter example
+      # that uses typed Kuberentes APIs
+      url: https://gist.github.com/cppforlife/48f41372cdc11dc7113f295377ef2074
+      ref: origin/main
+  template:
+  - cue:
+      outputExpression: "all"
+  deploy:
+  - kapp: {}
+```
+
+Internally, kapp-controller will just execute `cue export . --out yaml --expression all` to assemble configuration into one stream and pass it on to kapp.
+
+If you want to give above App CR a try, use [following RBAC setup](https://gist.githubusercontent.com/cppforlife/0dfde80e93933a62ae10a665baba64f8/raw/4f290fea6e7d07d1d4ce47d2b9ad11c5f6a9633d/config.yml) for simple-app-sa ServiceAccount referenced in `spec.serviceAccountName`.
+
+Once you've got a hang of App CR basics, it becomes pretty simple to take advantage of other generic kapp-controller features with Cue, for example, loading [Mozilla sops + age](https://github.com/mozilla/sops) encrypted environment specific configuration (stored in Git repository) into your Cue templates as inputs:
+
+```yaml
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: simple-app
+  namespace: default
+spec:
+  serviceAccountName: simple-app-sa
+  fetch:
+  - git:
+      url: https://gist.github.com/cppforlife/3506224cb7b681e283376cd061b5bfc8
+      ref: origin/main
+  template:
+  - sops:
+      age:
+        privateKeysSecretRef:
+          name: age-decrypt
+  - cue:
+      inputExpression: "config:"
+      outputExpression: "all"
+      valuesFrom:
+      - path: vals.yml
+  - kbld: {}
+  deploy:
+  - kapp: {}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: age-decrypt
+  namespace: default
+stringData:
+  key.txt: |
+    # public key: age1s3z9duz8c856y6qwtquhcqt6svu5pzctycvcz8nw08es2n59qffs7usgr3
+    AGE-SECRET-KEY-19QRN8ST7VH4TPXM6HFPGLAR69NZU2N6M4JG8YHAM4X47KHCZM8JSCQRCH9
+```
+
+Here is what will happen once above App CR is on the cluster:
+- first, Git repo is fetched
+- then, sops template step decrypts all files with `*.sops.yml` extension and just turns them into `*.yml` extension
+- next, Cue template step picks up decrypted configuration (`vals.yml`) and feeds it into Cue execution as a value under `config:` field (take a look at how [app.cue](https://gist.github.com/cppforlife/3506224cb7b681e283376cd061b5bfc8#file-app-cue-L3-L6) defines what can be accepted as input -- `#Config` defines `hello_msg` field must be a string, and no other keys are allowed)
+- [Carvel's kbld](/kbld) template step ensures that all container images are referenced by their digest
+- finally, kapp deploys produced resources
+
+These were two short and sweet examples of how to use Cue and kapp-controller together as part of your GitOps workflow, and perhaps we'll publish a follow-up post describing how to turn your App CR into a Package CR so that you can easily distribute your Cue templates as Carvel packages (with help of [Carvel's imgpkg](/imgpkg)) but for now -- that's a wrap.
```

Comment:
Just a thought: instead of teasing with a part 2, you could prompt the reader for feedback by suggesting they let us know if they want to see a packaging article. It's an invitation for a conversation, at least. 🤷🏻